lib/kubernetes-deploy/kubernetes_resource/pod.rb

```diff
@@ -4,50 +4,31 @@ class Pod < KubernetesResource
     TIMEOUT = 10.minutes
     SUSPICIOUS_CONTAINER_STATES = %w(ImagePullBackOff RunContainerError ErrImagePull).freeze
 
-    def initialize(name:, namespace:, context:, file:, parent: nil, logger:)
-      @name = name
-      @namespace = namespace
-      @context = context
-      @file = file
+    def initialize(namespace:, context:, definition:, logger:, parent: nil, deploy_started: nil)
       @parent = parent
-      @logger = logger
+      @deploy_started = deploy_started
+      @containers = definition["spec"]["containers"].map { |c| c["name"] }
```

Comment:
Nit: using `definition.fetch("spec")` instead of `definition["spec"]` would automatically raise if this key is not present (fail earlier), which is good because now when `spec` key is missing it will lead to `NoMethodError`